test/image/mocks/multicategory-inside-ticks.json

```diff
@@ -0,0 +1,76 @@
+{
+  "data": [
+    {
+      "type": "bar",
+      "x": [
+        ["2017", "2017", "2017", "2017", "2018", "2018", "2018"],
+        ["q1", "q2", "q3", "q4", "q1", "q2", "q3" ]
+      ],
+      "y": [1, 2, 3, 1, 3, 2, 3, 1]
+    },
+    {
+      "type": "bar",
+      "x": [
+        ["2017", "2017", "2017", "2017", "2018", "2018", "2018"],
+        ["q1", "q2", "q3", "q4", "q1", "q2", "q3"]
+      ],
+      "y": [1.12, 2.15, 3.07, 1.48, 2.78, 1.95, 2.54, 0.64]
+    },
+
+    {
+      "type": "bar",
+      "x": [
+        ["2017", "2017", "2017", "2017", "2018", "2018", "2018"],
+        ["q1", "q2", "q3", "q4", "q1", "q2", "q3"]
+      ],
+      "y": [1.12, 2.15, 3.07, 1.48, 2.78, 1.95, 2.54, 0.64],
+      "xaxis": "x2",
+      "yaxis": "y2"
+    },
+    {
+      "type": "bar",
+      "x": [
+        ["2017", "2017", "2017", "2017", "2018", "2018", "2018"],
+        ["q1", "q2", "q3", "q4", "q1", "q2", "q3" ]
+      ],
+      "y": [1, 2, 3, 1, 3, 2, 3, 1],
+      "xaxis": "x2",
+      "yaxis": "y2"
+    }
+  ],
+  "layout": {
+    "grid": {"rows": 2, "columns": 1, "pattern": "independent"},
+    "xaxis": {
+      "side": "top",
+      "title": "MULTI-CATEGORY",
+      "range": [-0.5, 7],
+      "showline": true,
+      "tickfont": {"size": 16},
+      "ticks": "inside",
+      "ticklen": 10,
+      "tickcolor": "red",
+      "tickwidth": 2,
+      "dividercolor": "blue",
+      "dividerwidth": 2
+    },
+    "xaxis2": {
+      "title": "MULTI-CATEGORY",
+      "range": [-1, 6.5],
+      "showline": true,
+      "tickfont": {"size": 16},
+      "ticks": "inside",
+      "ticklen": 10,
+      "tickcolor": "red",
+      "tickwidth": 2,
```

Comment:
I noticed `tickwidth` may be considered in the calculation of labels. However using a large number e.g. `tickwidth: 50` could force rotate the labels which might not be desirable.
![newplot 5](https://user-images.githubusercontent.com/33888540/50013609-a0eb9500-ff8f-11e8-84f8-1caa25dba22d.png)

Reply:
It _may_ no be desirable, but it is _intended_ for now.

https://github.com/plotly/plotly.js/blob/4f8628f4c7b8f0d604f1f6e7585e962428316cff/src/plots/cartesian/axes.js#L2517